crowdfunding/projects/permissions.py

```diff
@@ -0,0 +1,27 @@
+# added to make custom permission that only the owner of a project can edit it
+
+from rest_framework import permissions
+
+class IsOwnerOrReadOnly(permissions.BasePermission):
+    def has_object_permission(self, request, view, obj):
+        if request.method in permissions.SAFE_METHODS:
+            return True
+        return obj.owner == request.user
+
+class IsSupporterOrReadOnly(permissions.BasePermission):
+    def has_object_permission(self, request, view, obj):
+        if request.method in permissions.SAFE_METHODS:
+            return True
+        return obj.supporter == request.user
+
+class IsOwnProfile(permissions.BasePermission):
+    def has_object_permission(self, request, view, obj):
+        if request.method in permissions.SAFE_METHODS:
+            return True
+        return obj.id == request.user.id
+
+class IsCommenterOrReadOnly(permissions.BasePermission):
+    def has_object_permission(self, request, view, obj):
+        if request.method in permissions.SAFE_METHODS:
+            return True
+        return obj.commenter == request.user
```

Comment:
how great would it be if we had just decided to use a generic term for the user who created a data instance?
perhaps "created_by" then you wouldn't need all of these different permissions.
Really good job on doing all these permissions by the way.